### ISSUE TITLE
Is this project dead?

### ISSUE THREAD

**Issue by stephanlachnit** [FeralInteractive/gamemode#234]
I'm currently concerned that no-one really maintains this. I want to bring this up since the last reaction from Feral was about 2 months ago, and even then the last real code change was in January (Intel iGPU PR), before that the last commit was in October 2019.
Also, Issues like #216 or #221 (and many more) aren't closed, and issues like #215 are without any response from Feral.
Even worse is the situation for PRs, even the simplest PRs like #207, #208, #212, #218, #227 or #229 aren't merged.

/cc @mdiluz, @gicmo, @kakra, @aejsmith, @afayaz-feral

It would be quite sad to see this project just laying around and possibly dying. Has nobody repo access anymore except Feral? If so, and if Feral isn't interested anymore in maintaining this, I propose to fork gamemode under a new "gamemode-team", which at least maintains basic issues and PRs (like the ones notes above). The linux gaming community is big enough to keep this alive. What do you think?

**Comment by mdiluz** [comment]
I'll get in touch with contacts at Feral and discuss. I suspect the current pandemic is playing a factor too.

**Comment by stephanlachnit** [comment]
> I'll get in touch with contacts at Feral and discuss.

Thanks!

> I suspect the current pandemic is playing a factor too.

Oh yes that's a good point, I haven't considered that.

**Comment by gicmo** [comment]
Thanks for brining this up.


> It would be quite sad to see this project just laying around and possibly dying. Has nobody repo access anymore except Feral? If so, and if Feral isn't interested anymore in maintaining this, I propose to fork gamemode under a new "gamemode-team", which at least maintains basic issues and PRs (like the ones notes above). The linux gaming community is big enough to keep this alive. What do you think?

I am happy to help out reviewing patches, doing release and feature development (the latter to a lesser extent) if that helps. :man_technologist:

**Comment by afayaz-feral** [comment]
Sorry for not being so responsive, I've personally been juggling several things at once :)

As for this project, I'd given it less focus after Ubuntu/Fedora milestone freezes had gone by, with the intention of periodically looking at issues and PRs as well as making my own improvements. In particular, I'd have made sure to give the project focus again before a distro's major milestone.

This is obviously the wrong approach for a project with multiple frequent contributors outside of Feral. In the future, I'll dedicate some time to this project on a more regular basis.

There is definitely merit to opening up privileges to increase the bus factor, and we're currently in discussion internally about this.

**Comment by jqadev** [comment]
@afayaz-feral what do you think about creating a release candidate and a milestone of the next 1.6.0 version? The last one was created 2 years ago https://github.com/FeralInteractive/gamemode/milestones
It would be great to know when we can expect the next release and what is a blocker as there were merged a lot of changes since the last release.
It would also encourage people to try out a release candidate with various games before the final release to catch new issues or regressions as it is currently practiced in the Proton project.

**Comment by stephanlachnit** [comment]
A Milestone sounds like a good idea. I want to write a man page for `gamemode-simulate-game` before the next release, resolving #226 and #232 would be nice as well. But besides these minors things, there already have been some changes since 1.5.1, distributing them would be nice.
The Debian import freeze for Ubuntu 20.10 is on August 27th, so doing a release now and one in the beginning of August if there are any changes then sounds a like good idea to me, but in the end it depends on how much time @afayaz-feral has.

**Comment by afayaz-feral** [comment]
> @afayaz-feral what do you think about creating a release candidate and a milestone of the next 1.6.0 version?

I really like this idea, especially if it means we can get the community behind it and get some testing done out in the wild on a wide range of configs, and focus on just fixing bugs before a release.

> The Debian import freeze for Ubuntu 20.10 is on August 27th

We've effectively been doing time based releases already, typically in line with getting gamemode packaged in major distros. I'm happy to target the next milestone and RC for this.

PRs and issues tend to trickle in continuously, so it would be good to be able to draw a line and determine which should be resolved for the next release and which should be deferred.

I'm currently in the middle of reviewing the open PRs, but #228, #241 and #242 would definitely be good to have merged for 1.6. I'll have a look at open issues in the morning.

**Comment by afayaz-feral** [comment]
> I want to write a man page for gamemode-simulate-game before the next release

Definitely worth creating an issue for this, so I can add it to the 1.6 milestone. It's also worth just documenting it, in case someone else is able to pick it up.

**Comment by stephanlachnit** [comment]
> I really like this idea, especially if it means we can get the community behind it and get some testing done out in the wild on a wide range of configs, and focus on just fixing bugs before a release.

Speaking of that, when I have time I can go through some older issues which probably won't apply anymore (I a couple of Ubuntu 16.04 build problems), and ping you to close them. Right now I have the feeling that the issue tracker has more "I need help building this" than "This is a problem" or "This is a realistic feature".

> Definitely worth creating an issue for this, so I can add it to the 1.6 milestone. It's also worth just documenting it, in case someone else is able to pick it up.

That would be #244 then.
Some other issues I spotted that one could consider for 1.6:
- #246 easy, but depends on #176 
- #245 is quite easy to do
- #206 shouldn't be too hard
- #180 sounds nice but also like a lot of work
- #176 should definitely be fixed at some point
- #118 maybe, haven't look deeply into it

**Comment by stephanlachnit** [comment]
Just a little reminder, the Ubuntu feature freeze is on Aug 27 (https://wiki.ubuntu.com/GroovyGorilla/ReleaseSchedule). So still plenty of time, I think a new release in about a month would be a good idea.

**Comment by afayaz-feral** [comment]
Thanks for the heads up! I've been keeping an eye on schedules and I also think we should aim for early August.

**Comment by gicmo** [comment]
Fedora 33 will branch on `Tue 2020-08-11` and the code freeze is on `Tue 2020-08-25` ([F33 schedule](https://fedorapeople.org/groups/schedule/f-33/f-33-key-tasks.html)). Before the branch point would be great but at least for the code freeze (which aligns anyway with Ubuntu).

**Comment by stephanlachnit** [comment]
@afayaz-feral any news on the release? I guess we missed Ubuntu 20.10 (maybe they'll make an exception but I doubt it).

**Comment by afayaz-feral** [comment]
Sadly we have missed the distro freezes. I had been looking at the issues in the 1.6 milestone, and there were some issues still open. Although, #180 would have been a fairly huge change and a lot of work. I suppose the list of issues added to the 1.6 milestone was quite arbitrary.

There are a lot of small changes between 1.5.1 and master but each of those alone wouldn't warrant a release. Together, it does still make quite a nice update. I can do a release sometime next week, even if it's just to have a "1.6" release with bugfixes for people to update to separately to distro packaging.

I don't think having a milestone was a good idea for me - I held off doing a release because I wanted to close all of the issues first, even if that meant writing code for e.g. #180 myself. The last few weeks I have been very busy with my main responsibilities (specifically, making the list of games in README.md larger), so I just didn't have the time to dedicate to it.

I'm going to mark 1st Feb 2021 to do a release that week. That should be a couple of weeks before the Fedora 34 branch and a few weeks before the Ubuntu 21.04 feature freeze, I've estimated these looking at the schedules of Fedora 32 and Ubuntu 20.04. Whatever changes we happen to have then is what will get released.

**Comment by stephanlachnit** [comment]
@afayaz-feral just to let you know, the meson version wasn't updated for 1.6 (nothing dramatic).

**Comment by jqadev** [comment]
@afayaz-feral you forgot to bump version here https://github.com/FeralInteractive/gamemode/blob/master/meson.build#L5

**Comment by afayaz-feral** [comment]
@jqadev Thanks for the heads up! I'm currently updating the release script because the wrap file for inih creates a folder with a version suffix, which we don't currently handle.

**Comment by afayaz-feral** [comment]
Hi, thanks for the reminder! A new release has been on my todo list - I agree that it would be nice to have `gamemodelist` as well as the other PRs that have been merged since the last release.

I want to resolve the CI failure before that can happen, which looks like it's caused by a simple formatting error.

Unfortunately, due to some practical reasons I haven't been able to look into dealing with that or getting a new release ready. I'll see what I can do from here.

**Comment by stephanlachnit** [comment]
> I want to resolve the CI failure before that can happen, which looks like it's caused by a simple formatting error.
> 
> Unfortunately, due to some practical reasons I haven't been able to look into dealing with that or getting a new release ready. I'll see what I can do from here.

Fixed it in https://github.com/FeralInteractive/gamemode/pull/370, I hope this is allow you to tag a new release :)

The PRs https://github.com/FeralInteractive/gamemode/pull/357, https://github.com/FeralInteractive/gamemode/pull/360, https://github.com/FeralInteractive/gamemode/pull/362, https://github.com/FeralInteractive/gamemode/pull/368 and https://github.com/FeralInteractive/gamemode/pull/369 are all minor changes that can be merged as well.

**Comment by afayaz-feral** [comment]
@stephanlachnit I've now tagged a new release 1.7.

Thanks again for fixing the CI issue!
